packages/vite/src/node/plugins/resolve.ts

```diff
@@ -1117,35 +1109,31 @@ function packageEntryFailure(id: string, details?: string) {
 function resolveExportsOrImports(
   pkg: PackageData['data'],
   key: string,
-  options: InternalResolveOptionsWithOverrideConditions,
+  options: InternalResolveOptions,
   type: 'imports' | 'exports',
 ) {
-  const additionalConditions = new Set(
-    options.overrideConditions || [
-      'production',
-      'development',
-      'module',
-      ...options.conditions,
-    ],
+  const conditions = [...options.conditions, 'require', 'import'].filter(
+    (condition) => {
+      switch (condition) {
+        case 'production':
+          return options.isProduction
+        case 'development':
+          return !options.isProduction
+        case 'require':
+          return options.isRequire
+        case 'import':
+          return !options.isRequire
+        case 'node':
+          return !options.webCompatible
+        case 'browser':
+          return options.webCompatible
```

Comment:
I'm using `webCompatible` here for now. But probably we should do it in the env side. I left this for now to do that when removing the webCompatible flag. I think it would be easier to understand the diff later if it's done at the same time with the `browser` field.

Reply:
Update: I removed `webCompatible` in this PR (edcf8f3eca8e009382fdbc1a9e2a1771f8badca6)